### DISCUSSION TITLE
Filtering in relationship picker

### DISCUSSION THREAD

**Discussion by jjean-controlc** [Budibase/budibase#4596] [Help]
Hello budibase community,
Is it possible to filter the options of a relationship picker with in a form? In stead of having the choice of the entire dataset, I would like to have only a subset, defined in a view of the builtin database.

Thanks a lot

**Comment by mjashanks** [comment]
Hey @jjean-controlc !

I just answered this same question about an hour ago.... https://github.com/Budibase/budibase/discussions/4579 !

This one talked about ordering, but you can also apply a filter this way.

**Reply by jjean-controlc** [reply]
Hi @mjashanks 
Thanks a lot for the support and the quick answer
Indeed I was able to do it, but I'm still try to figure out how can I save that choice when saving the row in the form (my field remains empty).

**Reply by mjashanks** [reply]
Did you manually type the `Field` of the options picker as the exact same name as the relationship column?

**Comment by gerhardmpl** [comment]
@mjashanks How would that work with a many-to-many relationships? Tried the `multi-select picker` but got a _this field setting is the wrong data type for this component_ error message. I have a list of tasks marked as _active_ (boolean field) or not in a bbdb table and would like to filter on that when using a picker for the relationship in a related bbdb table.

**Reply by melohagan** [reply]
@gerhardmpl If you're working with relationships, you'll want to use the Relationship Picker:

<img width="153" alt="Screenshot 2022-04-15 at 16 43 20" src="https://user-images.githubusercontent.com/101575380/163591397-787ccfa9-e78b-40d5-84c0-4ed464f1673c.png">

This discussion might help as well: https://github.com/Budibase/budibase/discussions/5436
